pilot/pkg/config/kube/ingress/controller.go

```diff
@@ -186,29 +187,31 @@ func (c *controller) onEvent(item types.NamespacedName) error {
 	// we should check need process only when event is not delete,
 	// if it is delete event, and previously processed, we need to process too.
 	if event != model.EventDelete {
+		oldIngress := c.ingresses[config.NamespacedName(ing)]
 		shouldProcess := c.shouldProcessIngressUpdate(ing)
 		if !shouldProcess {
 			return nil
 		}
+		// only trigger when real changes were found
```

Comment:
This doesn't seem quite right because it doesn't take into account other dependencies? For example, when Service changes, we need to do a push. Even if Ingress doesn't change

Reply:
I also don't get why we can remove the AlwaysPush. The VS and GW we are pushing below have no spec, so won't they just always be skipped?

Reply:
as to `AlwaysPush`, we have annotation with `istio.io`

As to `This doesn't seem quite right because it doesn't take into account other dependencies?`

Thanks for remind, need to revert it. Seems we have no way to distinguish unless updating the input. Let's keep the original

Reply:
```
	for annotation := range curr.Meta.Annotations {
		if strings.Contains(annotation, "istio.io") {
			return true
		}
	}
``